src/VisualStudio/Core/Def/Utilities/IServiceProviderExtensions.cs

```diff
@@ -3,19 +3,18 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using System.Diagnostics;
 
 namespace Microsoft.VisualStudio.LanguageServices.Utilities
 {
     internal static class IServiceProviderExtensions
     {
-        /// <summary>
-        /// Returns the specified interface from the service. This is useful when the service and interface differ
-        /// </summary>
-        public static TInterfaceType GetService<TInterfaceType, TServiceType>(this IServiceProvider sp)
-            where TInterfaceType : class
-            where TServiceType : class
+        /// <inheritdoc cref="Shell.ServiceExtensions.GetService{TService, TInterface}(IServiceProvider, bool)"/>
```

Comment:
Any reason we're not just using the one that comes from the Shell binaries in the first place? Will this also inherit a <param> documentation that won't apply for whatever that bool parameter is?

Reply:
> Any reason we're not just using the one that comes from the Shell binaries in the first place?

It uses `ThreadHelper.JoinableTaskFactory`. I'm not sure we initialize that  property in test scenarios.

> Will this also inherit a documentation that won't apply for whatever that bool parameter is?

Not really, it just ignores it.